node.gyp

```diff
@@ -759,12 +824,44 @@
         } ],
       ]
     },
+    {
+      # When using shared lib to build executable in Windows, in order to avoid
+      # filename collision, the executable is node-win.exe. Need to rename it
+      # back to node.exe
```

Comment:
Was discussing this with Yihong last week, and I think it's worth pointing this out specifically.

In the Unix build, we build `libnode.so` (which gets linked as `-lnode`) and `node` (the binary).

In Windows, we can't build `node.dll` and `node.exe`, because the `node.pdb` and `node.lib` files clash. So this PR currently builds `node.exe` as `node-win.exe` to avoid the clash.

I guess the alternatives would be:
 1. Find some neat way to avoid the conflicts and still call it `node.exe` (sounds like this isn't possible, but maybe there's a way).
 2. Call them `node-win.dll` and `node.exe` (I think this is worse than what the PR does currently).

So assuming `1.` isn't possible, I'm leaning towards what this PR does, I guess the only downside is that the `dll` and `pdb` files will have the wrong names, maybe they should be renamed too?

Reply:
or keep using `node-win.exe` and modify the `test.py` to pick up the new name in windows platform for this case?

Reply:
If I understand correctly, the problem is the clash in the symbol name? Can you use /Fd with VC++to specify the pdb filename?